src/units/unit.hpp

```diff
@@ -1790,6 +1802,8 @@ class unit : public std::enable_shared_from_this<unit>
 private:
 
 	const std::set<std::string> checking_tags_{"attacks", "damage", "chance_to_hit", "berserk", "swarm", "drains", "heal_on_hit", "plague", "slow", "petrifies", "firststrike", "poison"};
+
+	const std::set<std::string> checking_halo_tags_{"heals", "hides", "illuminates", "leadership", "regenerate", "resistance", "skirmisher", "teleport", "attacks", "damage", "chance_to_hit", "berserk", "swarm", "drains", "heal_on_hit", "plague", "slow", "petrifies", "firststrike", "poison"};
```

Comment:
Having it be an ability named `[halo]` (and `[overlay]`) seems clearer to me than relying on a hardcoded list of other tag names. Is this added because priority was removed, and if so, why wasn't the method of filtering based on an ability being active able to work without priority?

I know that I asked for the attribute to be named `halo_image=` with the idea that we could put it into other tags, but that's something that I'd like to look at in 1.19 rather than the run-up to releasing 1.18.

Reply:
priority alreadyremoved

Reply:
> I know that I asked for the attribute to be named `halo_image=` with the idea that we could put it into other tags, but that's something that I'd like to look at in 1.19 rather than the run-up to releasing 1.18.

i pull https://github.com/wesnoth/wesnoth/pull/7856 for show you that what could be the code for this idea for 1.19